### PR TITLE
feature: add poetry support

### PR DESCRIPTION
**Pull request by steven-mi**

### CODE REVIEW

File: rocket/rocket.py [hunk 1]
```diff
@@ -107,18 +106,24 @@ def _build(self):
...
-        if not os.path.exists(f"{self.project_location}/setup.py"):
+        if os.path.exists(f"{self.project_location}/setup.py"):
+            # cleans up dist folder from previous build
+            dist_location = f"/tmp/db_rocket_builds"
+            self._shell(f"rm {dist_location}/* 2>/dev/null || true")
+            self._shell(
+                f"cd {self.project_location} ; {self._python_executable} -m build --outdir {dist_location}"
+            )
+        elif os.path.exists(f"{self.project_location}/pyproject.toml"):
+            dist_location = f"{self.project_location}/dist"
```
Comment (jeanCarloMachado) [inline rocket/rocket.py]: lets use the same dist location from both of them? Either place is fine for me.

Comment (jeanCarloMachado) [reply rocket/rocket.py]: Also the cleaning step, lets use the same :)

**Review by jeanCarloMachado — APPROVED** [review]
Thanks for open-sourcing :).

File: rocket/rocket.py (hunk 1 above)
Comment (jeanCarloMachado) [reply rocket/rocket.py]: Maybe should we introduce some test here? :)

Comment (steven-mi) [reply rocket/rocket.py]: With poetry you can't specify a location directory. It always creates the build files in the current folder. I'll update the python project folder to the poetry one.

Comment (steven-mi) [reply rocket/rocket.py]: > Maybe should we introduce some test here? :)

Yeah I'll create a new PR for this. I think the first tests will be a larger PR

**Pull request merged**